src/ConfiguredMediator.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/**
+ * Copyright by the ComposerDistributor-Team
+ *
+ * Licenses under the MIT-license. For details see the included file LICENSE.md
+ */
+
+declare(strict_types=1);
+
+namespace PharIo\ComposerDistributor;
+
+use Composer\Installer\PackageEvent;
+use PharIo\ComposerDistributor\Config;
+
+
+abstract class ConfiguredMediator extends PluginBase
+{
+	protected abstract function getMediatorConfig(): string;
+
+	public function installOrUpdateFunction(PackageEvent $event): void
+	{
+		$config    = Config\Factory::createFromFile($this->getMediatorConfig());
+		$installer = $this->createInstaller(
+			$config->package(),
+			$config->keyDirectory(),
+			$event
+		);
+
+		$files = [];
+		foreach ($config->phars() as $phar) {
+			$files[] = new File(
```

Comment:
Why do we need to create a new Class here? Can't we use the `Phar` class here instead? Or use a `File::createFromPharObject()` method? Or on the other hand use the `File` instead for the configuration?

Reply:
Good Question, I will investigate :)

Reply:
I removed the `Phar` class and replaced it with `File`and changed the return type from `Config::phars` to `FileList`